### DISCUSSION TITLE
Is there a way to save the training log as a txt file with mlx_lm

### DISCUSSION THREAD

**Discussion by Satyam7166-tech** [ml-explore/mlx-examples#939] [Q&A] [answered]
For example,

This is the command that I use

```
 mlx_lm.lora \
  --train \
  --model meta-llama/Meta-Llama-3-8B-Instruct \
  --data /Users/macstudiosct/projects/finetune_mlx/data \
  --batch-size 3 \
  --lora-layers -1 \
  --iters 5000 --max-seq-length 8000 --resume-adapter-file adapters/0002700_adapters.safetensors --adapter-path adapters_2

```
Here, the loss is printed at the terminal itself. If there could be a way to save it to a txt file as well, or plot a graph after the training, it would be easier to monitor the training.

**Comment by awni** [comment, marked as answer]
You could always pipe it to a file or use `tee` to have it in a file and on std out.

```bash
mlx_lm.lora  <...> | tee results.txt
```

Another option is to use the API instead of the CLI and pass custom callback (which you could use for example to send results to a wandb instance). Here's the [call back](https://github.com/ml-explore/mlx-examples/blob/main/llms/mlx_lm/tuner/trainer.py#L162-L183) base class. This [notebook](https://gist.github.com/awni/773e2a12079da40a1cbc566686c84c8f) has an example of using the API.

**Reply by Satyam7166-tech** [reply]
Ah thanks @awni for the simple solution.

And here I was making changes to the trainer.py and lora.py files.

Though I have understood more about mlx_lm in the process, so there's that :)

**Reply by Satyam7166-tech** [reply]
<img width="616" alt="image" src="https://github.com/user-attachments/assets/fc6f900d-7512-40c0-bad1-fb97aa9605d5">

Yep I chose the slightly complicated path but it was fun :-)
